src/main/scala/uk/gov/hmrc/ct/computations/CPQ117.scala

```diff
@@ -17,16 +17,28 @@
 package uk.gov.hmrc.ct.computations
 
 import uk.gov.hmrc.ct.box._
+import uk.gov.hmrc.ct.computations.losses.lossReform2017Applies
 import uk.gov.hmrc.ct.computations.retriever.ComputationsBoxRetriever
+import uk.gov.hmrc.ct.ct600.v3.retriever.AboutThisReturnBoxRetriever
 
 case class CPQ117(value: Option[Boolean])
   extends CtBoxIdentifier(name = "Did any of these losses arise from NIR activity?")
   with CtOptionalBoolean
   with Input
   with ValidatableBox[ComputationsBoxRetriever] {
 
-  override def validate(boxRetriever: ComputationsBoxRetriever): Set[CtValidation] = collectErrors(
-    requiredErrorIf( value.isEmpty && boxRetriever.cpQ17().isTrue ),
-    cannotExistErrorIf(value.nonEmpty && boxRetriever.cpQ17().isFalse )
-  )
+  override def validate(boxRetriever: ComputationsBoxRetriever): Set[CtValidation] = {
+    val b5 = boxRetriever match {
+      case a: AboutThisReturnBoxRetriever => a.b5().value.getOrElse(false)
```

Comment:
I don't like this cast. Test to cover this is needed.

Reply:
I will merge this to unblock the others, but please create tests as soon as possible